lib/compress/zstd_compress_internal.h

```diff
@@ -282,6 +282,7 @@ struct ZSTD_CCtx_s {
 #ifdef ZSTD_MULTITHREAD
     ZSTDMT_CCtx* mtctx;
 #endif
+    sem_t *semaphore;
```

Comment:
This should only be required when `ZSTD_MULTITHREAD` is defined.
Keep in mind that, in many systems, `libzstd` must be compiled within multithread support nor dependency.